dracut/30ignition/ignition-setup.sh

```diff
@@ -20,3 +26,5 @@ for src in ${sources[*]}; do
         done
     fi
 done
+
+umount $bootmnt
```

Comment:
umounting sometimes fails and needs a retry loop. See https://github.com/coreos/bootengine/blob/master/dracut/30ignition/retry-umount.sh

Reply:
I wonder if we could solve this by using a separate mount namespace like is done over here: 
https://github.com/coreos/ignition-dracut/blob/9e182111a768dd2f72ef3bb0917f96311915d8e6/systemd/ignition-firstboot-complete.service#L19-L20

Don't know if a separate mount namespace would mean it got unmounted after the unit was done or not.

Reply:
Looks like it works:

```
[root@vanilla-f29-atomic system]# ls /boot/
grub2  loader  loader.0  lost+found  ostree
[root@vanilla-f29-atomic system]# umount /boot
[root@vanilla-f29-atomic system]# ls /boot/
[root@vanilla-f29-atomic system]#
[root@vanilla-f29-atomic system]#
[root@vanilla-f29-atomic system]# systemctl daemon-reload
[root@vanilla-f29-atomic system]# systemctl cat foo.service
# /etc/systemd/system/foo.service
[Unit]
Description=foo

[Service]
Type=oneshot
RemainAfterExit=yes
MountFlags=slave
ExecStart=/bin/sh -c 'mount /boot && ls /boot/'

[Install]
[root@vanilla-f29-atomic system]#
[root@vanilla-f29-atomic system]# systemctl daemon-reload
[root@vanilla-f29-atomic system]# journalctl -u foo -f &
[2] 3169
[root@vanilla-f29-atomic system]# -- Logs begin at Thu 2019-02-14 03:55:04 UTC. --
Feb 14 03:57:59 vanilla-f29-atomic systemd[1]: /etc/systemd/system/foo.service:2: RequiresMountsFor= path is not absolute, ignoring: !/boot
Feb 14 03:57:59 vanilla-f29-atomic systemd[1]: /etc/systemd/system/foo.service:2: RequiresMountsFor= path is not absolute, ignoring: !/boot

[root@vanilla-f29-atomic system]#
[root@vanilla-f29-atomic system]#
[root@vanilla-f29-atomic system]# systemctl start foo.service
[root@vanilla-f29-atomic system]# Feb 14 03:59:36 vanilla-f29-atomic systemd[1]: Starting foo...
Feb 14 03:59:36 vanilla-f29-atomic systemd[1]: Starting foo...
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: grub2
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: loader
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: grub2
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: loader.0
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: lost+found
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: loader
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: ostree
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: loader.0
Feb 14 03:59:36 vanilla-f29-atomic systemd[1]: Started foo.
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: lost+found
Feb 14 03:59:36 vanilla-f29-atomic sh[3172]: ostree
Feb 14 03:59:36 vanilla-f29-atomic systemd[1]: Started foo.

[root@vanilla-f29-atomic system]# ls /boot/
[root@vanilla-f29-atomic system]#


```

I'll add this to this PR